tests/python/gaia-ui-tests/gaiatest/tests/functional/messages/test_sms_report.py

```diff
@@ -0,0 +1,86 @@
+# This Source Code Form is subject to the terms of the Mozilla Public
+# License, v. 2.0. If a copy of the MPL was not distributed with this
+# file, You can obtain one at http://mozilla.org/MPL/2.0/.
+
+import time
+from marionette_driver import expected, By, Wait
+
+from gaiatest import GaiaTestCase
+from gaiatest.apps.base import Base
+from gaiatest.apps.messages.app import Messages
+
+
+class TestSmsReport(GaiaTestCase):
+
+    time_sent = time.time()
+
+    def test_sms_report(self):
+        """This test sends a text message to itself. It waits for a reply message.
+           Then it checks the info in the report.
+        """
+        _text_type = (By.CSS_SELECTOR, 'p[data-l10n-id="message-type-sms"]')
+        _sender = (By.CSS_SELECTOR, 'a[data-name]')
+        _receiver = (By.CSS_SELECTOR, 'a.suggestion.js-contact-info')
+
+        
+        # Configuring the test
+        _text_message_content = "Automated Test %s" % str(self.time_sent)
+        
+        _test_phone_number = self.testvars['phone_numbers']
+        _text_type_expected = "Text Message (SMS)"
+        
+        # Sending the message
+        self.messages = Messages(self.marionette)
+        self.messages.launch()
+
+        new_message = self.messages.create_new_message(recipients=_test_phone_number,
+                                                       message=_text_message_content)
+        self.message_thread = new_message.tap_send()
+        self.message_thread.wait_for_received_messages()
+
+        last_received_message = self.message_thread.received_messages[-1]
+        last_message = self.message_thread.all_messages[-1]
+
+        self.assertEqual(_text_message_content, last_received_message.text)
+        self.assertEqual(last_received_message.id, last_message.id)
+        
+        # Long Tap on received message and check the report: SMS sent received and time
+        activities = last_received_message.long_press_message()
+        time.sleep(5) # Will try to change this
```

Comment:
I see why you want to wait 5 seconds, and I surprised we've never had any issue before!

You need to add the correct wait in `long_press_message()`. When an activity picker start, it goes from the bottom to the top of the screen. You need to wait until it reaches that top. To do so, you'll have to change that function:
```py
class Activities(Base): # No need to change this line
    def __init__(self, marionette): # No need to change this line
        Base.__init__(self, marionette) # No need to change this line
        action_menu = Wait(self.marionette).until(expected.element_present(
                *self._actions_menu_locator))
        Wait(self.marionette).until(lamba m: action_menu.rect['y'] == 0)
```